### DISCUSSION TITLE
An app stuck probably due to some suspense related bug

### DISCUSSION THREAD

**Discussion by RIP21** [urql-graphql/urql#3262] [Q&A]
In development this error is not occurring but for production build in some 100% reproducible (appearing 100% of the time, not random) scenarios app just completely stucks on redirect which potentially suspenses some big chunk of the UI. 
As if URQL or React gets into an infinite loop. Browser cursor stuck, dev tools profiling cannot get the results back of the profiling etc.
Tab becomes completely non-responsive (dev tools are responsive, but all UI is not) as you cannot refresh the page or navigate using buttons.

At the same time, If you change the sequence of actions to open something different, then return and click on the same thing, it works just fine.

It makes me believe suspense is VERY unstable still. And what's worse is that it only happens on production builds, there is no way to reproduce that in development.

**Comment by RIP21** [comment]
What's worse is that it still happening that some of the queries are failing with field access on `data` even tho, you said to me another day that it should only occur on development, not a production build. But it still happening.

I assume it's safe to say that suspense is rather not production ready or much harder than "just wrap some stuff in Suspense".
I think I'll have to rewrite all my changes to be using non suspended version, unfortunately. 

For reference, something similar happened to other Next.js experiments I did that were literally 2-3 files with some queries and still I've encountered similar freezes in production builds (in development too, but I blame Jotai (jotai-urql binding that I've built, it only happened with it and not URQL default bindings, but probably as I got lucky), not URQL in this case). So maybe it's Next.js + URQL that is not working well, maybe something else, but I definitely struggle with it a lot when using suspense...

**Reply by kitten** [reply]
Do you have any reproductions or specific cases applying to this? I'm not aware of any cases where data can leak without suspense, but some may exist 

That said, what I was referring to was that there are multiple cases in which a valid `data: null` result can be the result of a query, suspense or not.

This is because whenever an error bubbles up, `data` would be `null`, however, this can also happen with partial/stale loading results in Graphcache.

Anyway, hard to tell what the conditions for your error are at the top of my head.
Either way, it's not a good idea to rely on `data` not being `null` in all cases, even with suspense and a manual error handler.

In development, hot module reloading can easily cause this case as well, afaik, temporarily
